### PR TITLE
Bug #1915 - Remove AddMvc overload with default parameter.

### PR DESCRIPTION
**Pull request by sornaks**
Linking #1915

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -13,7 +13,15 @@ namespace Microsoft.Framework.DependencyInjection
 {
     public static class MvcServiceCollectionExtensions
     {
-        public static IServiceCollection AddMvc(this IServiceCollection services, IConfiguration configuration = null)
+        public static IServiceCollection AddMvc([NotNull]this IServiceCollection services)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: spacing

File: src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -13,7 +13,15 @@ namespace Microsoft.Framework.DependencyInjection
...
+        {
+            ConfigureDefaultServices(services, null);
+            services.TryAdd(MvcServices.GetDefaultServices(null));
+            return services;
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: should this just be `return AddMvc(services, null)`?

File: src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -13,7 +13,15 @@ namespace Microsoft.Framework.DependencyInjection
...
+        }
+
+        public static IServiceCollection AddMvc(
+            [NotNull]this IServiceCollection services, [NotNull]IConfiguration configuration)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: spacing, put these each on their own line

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: Why is configuration `[NotNull]`? Clearly it's allowed to be null from reading the code

**Comment by rynowak** [comment]
:watch:

File: src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs (hunk 3 above)
Comment (sornaks) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: Yes, it is allowed to be null (which is covered by the 1st overload). @yishaigalatzer wanted an overload which takes a `NotNull` configuration.

Comment (sornaks) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: Discussed with Yishai. He does not have a strong opinion on this. Will remove `NotNull` and make the above overload call this.

**Comment by sornaks** [comment]
Updated.. Please have a look..

File: src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -13,7 +13,14 @@ namespace Microsoft.Framework.DependencyInjection
 {
     public static class MvcServiceCollectionExtensions
     {
-        public static IServiceCollection AddMvc(this IServiceCollection services, IConfiguration configuration = null)
+        public static IServiceCollection AddMvc([NotNull] this IServiceCollection services)
+        {
+            return AddMvc(services, null);
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: named parameter:

`return AddMvc(services, configuration: null);`

**Comment by rynowak** [comment]
:shipit: from me

**Comment by sornaks** [comment]
Checked in. Thanks! https://github.com/aspnet/Mvc/commit/accf0c76f0bbcca71a1b2faa3532f021a29532fe

**Pull request closed without merging**